lib/compress/index.js

```diff
@@ -6941,14 +6942,37 @@ AST_Lambda.DEFMETHOD("contains_this", function() {
     });
 });
 
+const objectPrototypeProperties = new Set([
+    "__count__",
+    "__defineGetter__",
+    "__defineSetter__",
+    "__lookupGetter__",
+    "__lookupSetter__",
+    "__noSuchMethod__",
+    "__parent__",
+    "__proto__",
+    "constructor",
+    "eval",
+    "hasOwnProperty",
+    "isPrototypeOf",
+    "propertyIsEnumerable",
+    "toLocaleString",
+    "toSource",
+    "toString",
+    "unwatch",
+    "valueOf",
+    "watch",
+]);
 AST_PropAccess.DEFMETHOD("flatten_object", function(key, compressor) {
     if (!compressor.option("properties")) return;
     var arrows = compressor.option("unsafe_arrows") && compressor.option("ecma") >= 2015;
     var expr = this.expression;
     if (expr instanceof AST_Object) {
         var props = expr.properties;
+        var collapse_missing_props = compressor.option("unsafe");
         for (var i = props.length; --i >= 0;) {
             var prop = props[i];
+            if (prop.value instanceof AST_Accessor) collapse_missing_props = false;
```

Comment:
I don't think there's a need for this condition here. If any of the properties is an accessor, it can still be doomed to oblivion :)